frontend/packages/ceph-storage-plugin/src/components/modals/storage-class-dropdown.tsx

```diff
@@ -14,18 +15,30 @@ export const OCSStorageClassDropdown: React.FC<OCSStorageClassDropdownProps> = (
 
 const StorageClassDropdown = (props: any) => {
   const scConfig = _.cloneDeep(props);
+  const [infrastructure, setInfrastructure] = React.useState<K8sResourceKind>();
+  const [infrastructureError, setInfrastructureError] = React.useState();
   /* 'S' of Storage should be Capital as its defined key in resourses object */
   const scLoaded = _.get(scConfig.resources.StorageClass, 'loaded');
-  const scData = _.get(scConfig.resources.StorageClass, 'data', []) as K8sResourceKind[];
+  const scData = _.get(scConfig.resources.StorageClass, 'data', []) as StorageClassResourceKind[];
+
+  React.useEffect(() => {
+    const fetchInfrastructure = async () => {
+      try {
+        const infra = await k8sGet(InfrastructureModel, 'cluster');
+        setInfrastructure(infra);
+      } catch (error) {
+        setInfrastructureError(error);
+      }
+    };
+    fetchInfrastructure();
+  }, []);
 
-  const filteredSCData = scData.filter((sc: K8sResourceKind) => {
-    return cephStorageProvisioners.every(
-      (provisioner: string) => !_.get(sc, 'provisioner').includes(provisioner),
-    );
-  });
+  const infrastructurePlatform = getInfrastructurePlatform(infrastructure);
 
-  if (scLoaded) {
-    scConfig.resources.StorageClass.data = filteredSCData;
+  if (scLoaded && !infrastructureError && !!infrastructurePlatform) {
+    // find infra supported provisioner
+    const provisioner = infraProvisionerMap[_.toLower(infrastructurePlatform)];
+    scConfig.resources.StorageClass.data = _.filter(scData, (sc) => sc.provisioner === provisioner);
```

Comment:
Please add types for both `provisioner` and ` scConfig`

Reply:
Ack on provisioner. scConfig type is inherited.